src/renderer/components/wallet/settings/Settings.tsx

```diff
@@ -48,16 +48,30 @@ export const Settings: React.FC<Props> = (props): JSX.Element => {
 
   const addDevice = useCallback(
     (chain: Chain) => {
-      retrieveLedgerAddress({ chain })
+      switch (selectedNetwork) {
+        case 'mainnet':
+        case 'testnet':
+          retrieveLedgerAddress({ chain, network: selectedNetwork })
+          break
+        default:
+          break
+      }
     },
-    [retrieveLedgerAddress]
+    [retrieveLedgerAddress, selectedNetwork]
   )
 
   const removeDevice = useCallback(
     (chain: Chain) => {
-      removeLedgerAddress(chain)
+      switch (selectedNetwork) {
+        case 'mainnet':
+        case 'testnet':
+          removeLedgerAddress({ chain, network: selectedNetwork })
+          break
+        default:
+          break
+      }
```

Comment:
No check of network needed (see previous comments about network + `removeLedgerAddress`)